core/src/main/kotlin/io/zeebe/bpmnspec/builder/SpecBuilder.kt

```diff
@@ -0,0 +1,141 @@
+        val actionsBuilder = ActionsBuilder();
+
+        actionsBuilder.init()
+
+        actions.addAll(actionsBuilder.build())
+    }
+
+    fun verifications(init: VerificationsBuilder.() -> Unit) {
+        val verificationsBuilder = VerificationsBuilder();
+
+        verificationsBuilder.init()
+
+        verifications.addAll(verificationsBuilder.build())
+    }
+
+    fun build() : TestCase {
+        return TestCase(name, description, actions, verifications)
+    }
+}
+
+@TestSpecMarker
+class ActionsBuilder {
+    private val actions = ArrayList<Action>()
+
+    fun awaitElementInstanceState(selector: ElementSelector, state: ElementInstanceState, workflowInstance: String? = null) {
+        actions.add(AwaitElementInstanceStateAction(state, selector.elementId, selector.elementName, workflowInstance))
+    }
+
+    fun cancelInstance(workflowInstance: String? = null) {
+        actions.add(CancelInstanceAction(workflowInstance));
+    }
+
+    fun completeTask(jobType : String, variables: Map<String, Any> = Collections.emptyMap()) {
+        actions.add(CompleteTaskAction(jobType, serializeToJSON(variables)))
+    }
+
+    fun createInstance(bpmnProcessId: String, variables: Map<String, Any> = Collections.emptyMap(), workflowInstanceAlias: String? = null ) {
+        actions.add(CreateInstanceAction(bpmnProcessId, serializeToJSON(variables), workflowInstanceAlias))
+    }
+
+    fun publishMessage(messageName: String, correlationKey: String, variables: Map<String, Object> = Collections.emptyMap()) {
+        actions.add(PublishMessageAction(messageName, correlationKey, serializeToJSON(variables)))
+    }
+
+    fun throwError(jobType: String, errorCode: String, errorMessage: String = "" ) {
+        actions.add(ThrowErrorAction(jobType, errorCode, errorMessage));
+    }
+
+    fun build(): List<Action> {
+        return actions;
+    }
+}
+
+@TestSpecMarker
+class VerificationsBuilder {
+    private val verifications = ArrayList<Verification>()
+
+    fun elementInstanceState(selector: ElementSelector, state: ElementInstanceState, workflowInstance: String? = null) {
+        verifications.add(ElementInstanceStateVerification(state, selector.elementId, selector.elementName, workflowInstance))
+    }
+
+    fun workflowInstanceState(state: WorkflowInstanceState, workflowInstance: String? = null) {
+        verifications.add(WorkflowInstanceStateVerification(state, workflowInstance))
+    }
+
+    fun workflowInstanceVariable(variableName: String, variableValue: String, selector: ElementSelector, workflowInstance: String? = null) {
+        verifications.add(WorkflowInstanceVariableVerification(variableName, variableValue, workflowInstance, selector.elementId, selector.elementName))
+    }
+
+    fun workflowInstanceVariables(variables : Map<String, String>, selector: ElementSelector, workflowInstance: String? = null) {
+        variables.forEach { (key, value) -> workflowInstanceVariable(key, value, selector, workflowInstance) };
+    }
```

Comment:
The parameter `variables` is not aligned with the same parameter for the `createInstance` or `completeTask` action. Here, we need to specify the variable values as strings instead of objects. See the following examples:

```
testSpec {
                    resources("demo.bpmn")
                    testCase(name = "complete workflow", description = "demo test case") {
                        actions {
                            createInstance(bpmnProcessId = "demo")
                            completeTask(jobType = "a", variables = mapOf("x" to "foo", "y" to 1))
                            completeTask(jobType = "b")
                            completeTask(jobType = "c")
                        }
                        verifications {
                            elementInstanceState(selector = byName("A"), state = ElementInstanceState.COMPLETED )
                            workflowInstanceVariable(variableName = "x", variableValue = "\"foo\"")
                            workflowInstanceVariable(variableName = "y", variableValue = "1")
                            workflowInstanceVariables(variables = mapOf("x" to "\"foo\"", "y" to "1"))
                            workflowInstanceState(state = WorkflowInstanceState.COMPLETED)
                        }
                    }
                }
```

Can we allow objects instead of strings?